solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -1111,6 +1153,22 @@ contract StandardPoolConverter is
         return _targetReserveBalance.mul(_amount) / _sourceReserveBalance.add(_amount);
     }
 
+    function crossReserveSourceAmount(
+        uint256 _sourceReserveBalance,
+        uint256 _targetReserveBalance,
+        uint256 _amount
+    ) private pure returns (uint256) {
+        // validate input
+        require(_sourceReserveBalance > 0, "ERR_INVALID_RESERVE_BALANCE");
+        require(_amount < _targetReserveBalance, "ERR_INVALID_AMOUNT");
```

Comment:
we're comparing amount in source token with the target reserve balance - mistake?